Unable to use MQTT home assistant autodiscovery with homie autodiscovery disabled

As of 2021.4, when I have both homie and home assistant autodiscovery enabled, my vacuum is autodiscovered and shows up in home assistant. When I disable homie autodiscovery, but leave home assistant enabled, the device disappears from home assistant as if the autodiscovered device was deleted (even though I have "Delete autodiscovery on shutdown" disabled for both homie and home assistant). Looking in my logs, I see several messages that look like either:
```
[2021-04-26T01:49:57.528Z] [WARN] MQTT handle vacuum/robot/BatteryStateAttribute failed to configure Error: Handle may only be configured while the MQTT controller is not initialized
    at BatteryStateMqttHandle.configure (/snapshot/Valetudo/lib/mqtt/handles/MqttHandle.js:160:19)
    at BatteryStateMqttHandle.configure (/snapshot/Valetudo/lib/mqtt/handles/RobotStateNodeMqttHandle.js:30:21)
    at RobotMqttHandle.configure (/snapshot/Valetudo/lib/mqtt/handles/MqttHandle.js:167:29)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /snapshot/Valetudo/lib/mqtt/handles/RobotMqttHandle.js:110:13
    at async MqttController.reconfigure (/snapshot/Valetudo/lib/mqtt/MqttController.js:497:13)
    at async RobotMqttHandle.onStatusAttributeEvent (/snapshot/Valetudo/lib/mqtt/handles/RobotMqttHandle.js:97:9)
```
and
```
[2021-04-26T01:50:33.473Z] [ERROR] MQTT publication error: TypeError: Cannot read property 'publish' of null
    at MqttController.publishHass (/snapshot/Valetudo/lib/mqtt/MqttController.js:698:36)
    at HassController.refreshAutoconf (/snapshot/Valetudo/lib/mqtt/homeassistant/HassController.js:180:31)
    at InLineHassComponent.refreshAutoconf (/snapshot/Valetudo/lib/mqtt/homeassistant/components/HassComponent.js:146:25)
    at CallbackHassAnchorSubscriber.callback (/snapshot/Valetudo/lib/mqtt/homeassistant/components/HassComponent.js:23:24)
    at CallbackHassAnchorSubscriber.onAnchorPost (/snapshot/Valetudo/lib/mqtt/homeassistant/CallbackHassAnchorSubscriber.js:18:20)
    at HassAnchor.post (/snapshot/Valetudo/lib/mqtt/homeassistant/HassAnchor.js:62:34)
    at new InLineHassComponent (/snapshot/Valetudo/lib/mqtt/homeassistant/components/InLineHassComponent.js:24:40)
    at /snapshot/Valetudo/lib/mqtt/capabilities/WifiConfigurationCapabilityMqttHandle.js:85:17
    at MqttController.withHass (/snapshot/Valetudo/lib/mqtt/MqttController.js:676:13)
    at new WifiConfigurationCapabilityMqttHandle (/snapshot/Valetudo/lib/mqtt/capabilities/WifiConfigurationCapabilityMqttHandle.js:83:25)
```

## To Reproduce
1. Setup MQTT with homie and home assistant autodiscovery enabled, see device in home assistant, Valetudo logs:
```
[2021-04-26T01:50:33.803Z] [INFO] Connected successfully to MQTT broker
[2021-04-26T01:50:35.648Z] [INFO] MQTT configured
```
2. Disable homie autodiscovery and save, observe Valetudo logs:
```
[2021-04-26T02:01:29.401Z] [ERROR] MQTT publication error: TypeError: Cannot read property 'publish' of null
<repeated for 8 separate capabilities>
[2021-04-26T02:01:30.138Z] [INFO] Connected successfully to MQTT broker
[2021-04-26T02:01:31.981Z] [INFO] MQTT configured
```
In home assistant, the device is deleted.
3. Enable homie autodiscovery and save, observe vacuum reappear in home assistant.

### Vacuum Model
S4 Max

### Valetudo Version
2021.04.0

--- a/lib/mqtt/MqttController.js
+++ b/lib/mqtt/MqttController.js
@@ -438,7 +438,7 @@ class MqttController {
      * @return {Promise<void>}
      */
     async setState(state) {
-        if (this.homieEnabled) {
+        if (this.homieEnabled || this.hassEnabled) {
             await this.asyncClient.publish(this.stateTopic, state, {
                 // @ts-ignore
                 qos: MqttCommonAttributes.QOS.AT_LEAST_ONCE,

